src/structures/Message.js

```diff
@@ -541,18 +541,22 @@ class Message extends Base {
 
   /**
    * Edits the content of the message.
-   * @param {?(string|APIMessage)} [content] The new content for the message
-   * @param {MessageEditOptions|MessageEmbed|MessageAttachment|MessageAttachment[]} [options] The options to provide
+   * @param {string|APIMessage|MessageEditOptions|MessageEmbed
+   * |MessageAttachment|MessageAttachment[]} options The options to provide
    * @returns {Promise<Message>}
    * @example
    * // Update the content of a message
    * message.edit('This is my new content!')
    *   .then(msg => console.log(`Updated the content of a message to ${msg.content}`))
    *   .catch(console.error);
    */
-  edit(content, options) {
-    options = content instanceof APIMessage ? content : APIMessage.create(this, content, options);
-    return this.channel.messages.edit(this.id, options);
+  edit(options) {
+    let apiMessage;
+
+    if (options instanceof APIMessage) apiMessage = options.resolveData();
+    else apiMessage = APIMessage.create(this, options).resolveData();
+
+    return this.channel.messages.edit(this.id, apiMessage);
   }
```

Comment:
```suggestion
  edit(options) {
    return this.channel.messages.edit(this.id, options);
  }
```
The additional logic here was added in #5612, not sure if it is still needed after this PR but I believe not